querybook/webapp/redux/queryEngine/selector.ts

```diff
@@ -42,16 +50,32 @@ export const queryEngineByIdEnvSelector = createSelector(
         })
 );
 
+export const enabledQueryEngineByIdEnvSelector = createSelector(
+    enabledQueryEngineSelector,
+    (queryEngines) =>
+        new Proxy(arrayGroupByField(queryEngines), {
+            get: (engineById, name: string) =>
+                name in engineById ? engineById[name] : unknownQueryEngine,
+        })
+);
+
 export const queryEngineStatusAndEngineIdsSelector = createSelector(
     queryEngineStatusByIdSelector,
     engineIdsInEnvironmentSelector,
-    (queryEngineStatusById, engineIds) =>
-        engineIds.reduce((pairs, id) => {
-            if (id in queryEngineStatusById) {
-                pairs.push([id, queryEngineStatusById[id]]);
-            }
-            return pairs;
-        }, []) as Array<[number, IQueryEngineStatus]>
+    enabledQueryEngineSelector,
```

Comment:
update fetchAllSystemStatus to not fetch disabled engines, then the filter would be no longer needed?

Reply:
Good catch, I forgot to update that function.  But this filter is still needed to avoid showing all clusters in the status popover.